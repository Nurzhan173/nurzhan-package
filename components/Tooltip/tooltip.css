:root {
  --tooltip-top: 100%;
  --tooltip-transform: translateX(-50%);
}

#tooltip {
  position: relative;
  padding: 7px;

  cursor: pointer;
  font-size: 25px;
  font-weight: bold;
  font-family: sans-serif;
}

#tooltipText {
  position: absolute;
  left: 50%;
  top: 0;
  transform: translateX(-50%);

  background: #000;
  color: #fff;

  white-space: nowrap;

  padding: 10px 15px;
  border-radius: 7px;

  visibility: hidden;
  opacity: 0;
  transition: opacity 0.5s ease;
}

#tooltipText::before {
  content: "";
  position: absolute;
  left: 50%;
  top: var(--tooltip-top);
  transform: var(--tooltip-transform);
  border: 15px solid;
  border-color: #000 #0000 #0000 #0000;
}

#tooltip:hover #tooltipText {
  top: -30%;
  visibility: visible;
  opacity: 1;
}
