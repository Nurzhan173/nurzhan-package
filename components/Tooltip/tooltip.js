class Tooltip {

  constructor() {
    window.addEventListener('scroll', function() {
      const tooltip = document.getElementById('tooltip');
      const tooltipText = document.getElementById('tooltipText');
      const tooltipRect = tooltip.getBoundingClientRect();

      if(tooltipRect.top < 50) {
        tooltipText.style.top = "130%";
        tooltipText.style.setProperty("--tooltip-top", "-60%");
        tooltipText.style.setProperty("--tooltip-transform", "translateX(-50%) rotate(180deg)");
      } else if (tooltipRect.top > -50) {
        tooltipText.style.top = "-130%";
        tooltipText.style.setProperty("--tooltip-top", "100%");
        tooltipText.style.setProperty("--tooltip-transform", "translateX(-50%)");
      }
    });
  }

  render() {
    const tooltip = document.createElement('div');
    const tooltipText = document.createElement('span');

    tooltip.id = 'tooltip';
    tooltipText.id = 'tooltipText';

    tooltip.appendChild(tooltipText);

    return tooltip;
  }
}

module.exports = Tooltip;
