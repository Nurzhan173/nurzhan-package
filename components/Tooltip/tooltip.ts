import './tooltip.css';
/**
 * @module Tooltip
 */

/**
 * Need to create div element with id #tooltip
 * for text inside tooltip need to create div with id #tooltipText
 * @type {HTMLElement} tooltip - tooltip div wrapper
 * @type {HTMLElement} tooltipText - tooltip text
 * @returns {HTMLElement} returns tooltip
 */

class Tooltip {

  constructor() {
    window.addEventListener('scroll', function() {
      const tooltip = document.getElementById('tooltip');
      const tooltipText = document.getElementById('tooltipText');
      const tooltipRect = tooltip.getBoundingClientRect();

      if(tooltipRect.top < 50) {
        tooltipText.style.top = "130%";
        tooltipText.style.setProperty("--tooltip-top", "-60%");
        tooltipText.style.setProperty("--tooltip-transform", "translateX(-50%) rotate(180deg)");
      } else if (tooltipRect.top > -50) {
        tooltipText.style.top = "-130%";
        tooltipText.style.setProperty("--tooltip-top", "100%");
        tooltipText.style.setProperty("--tooltip-transform", "translateX(-50%)");
      }
    });
  }

  render() {
    const tooltip = document.createElement('div');
    const tooltipText = document.createElement('span');

    tooltip.id = 'tooltip';
    tooltipText.id = 'tooltipText';

    tooltip.appendChild(tooltipText);

    return tooltip;
  }
}

export default Tooltip;
