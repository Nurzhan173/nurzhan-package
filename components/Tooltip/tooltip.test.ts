import Tooltip from './tooltip';
import { describe, expect, beforeEach, afterEach, test } from '@jest/globals'


describe('Tooltip', () => {
  let tooltip;

  beforeEach(() => {
    // Create a new instance of Tooltip before each test
    tooltip = new Tooltip();
  });

  afterEach(() => {
    // Clean up any elements or listeners after each test
    document.body.innerHTML = '';
  });

  test('Tooltip renders a div element with the correct id', () => {
    const renderedTooltip = tooltip.render();
    expect(renderedTooltip.tagName).toBe('DIV');
    expect(renderedTooltip.id).toBe('tooltip');
  });

  // You can add more tests here to cover other aspects of your Tooltip class
  test('Tooltip text is initially hidden', () => {
    const renderedTooltip = tooltip.render();
    const tooltipText = renderedTooltip.querySelector('#tooltipText');
    expect(tooltipText.style.display).toBe('');
  });
});
