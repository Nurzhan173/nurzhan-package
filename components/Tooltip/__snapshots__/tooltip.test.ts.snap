// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Tooltip renders a div element with the correct id 1`] = `Tooltip {}`;
